File: scripts/reference-solutions/driver-rebalancing.cpp
#include <cstdlib>
#include <iostream>
#include <vector>
using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  int zones;
  cin >> zones;
  vector<long long> current(zones);
  vector<long long> target(zones);
  for (long long& drivers : current) cin >> drivers;
  for (long long& drivers : target) cin >> drivers;

  long long imbalance = 0;
  long long cost = 0;
  for (int index = 0; index + 1 < zones; ++index) {
    imbalance += current[index] - target[index];
    cost += llabs(imbalance);
  }

  cout << cost << '\n';
  return 0;
}

File: scripts/reference-solutions/driver-rebalancing.py
import sys


def main():
    data = list(map(int, sys.stdin.buffer.read().split()))
    zones = data[0]
    current = data[1 : zones + 1]
    target = data[zones + 1 : 2 * zones + 1]

    imbalance = 0
    cost = 0
    for index in range(zones - 1):
        imbalance += current[index] - target[index]
        cost += abs(imbalance)

    print(cost)


main()